### ISSUE TITLE
Cannot define an arrow function with a single generic type parameter that has a default value with JSX

### ISSUE THREAD

**Issue by giladgo** [microsoft/TypeScript#47062]
# Bug Report

### 🔎 Search Terms
typescript
jsx
tsx
generic
arrow function
default value

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about generics.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.5.2#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wChS0IA7AZ3hAE8AxAVyrTgF44AeAFS5w6UYFQDmAGgB8ACgCUXKXADe5AL5A)

### 💻 Code

```ts
import React from 'react'

const myFunc = <T = string,>() => {

}
```

### 🙁 Actual behavior

The typescript compiler throws multiple errors

### 🙂 Expected behavior

I would be allowed to create an arrow function with a _single_ type parameter that has a default value.

Note:
* This happens only on with JSX enabled
* Removing the default value (`= string`) doesn't throw errors
* Adding a second type parameter before `T`, ie `<K, T=string>` doesn't throw errors

**Comment by MartinJohns** [comment]
Duplicate of #15713.

Workaround: `T extends any = string`

**Comment by RyanCavanaugh** [comment]
We should lookahead one token here for `=` the same way we do for `,`

**Comment by giladgo** [comment]
Amazing! thanks :)

**Comment by florianmatz** [comment]
I'm to dumb to use it. @giladgo Could you provide a example syntax?

**Comment by AndreaEsposit** [comment]
I am experiencing the same problem on Typescript 5.1.3. Is it just me?
